addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -43,56 +52,77 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                 if (not has_header and not included_product_docs and not has_footer):
                     continue
 
+                all_form_fields = set()
                 writer = PdfFileWriter()
+
                 if has_header:
-                    self._add_pages_to_writer(writer, base64.b64decode(header_record.sale_header))
+                    decoded_header = base64.b64decode(header_record.sale_header)
+                    self._add_pages_to_writer(writer, decoded_header, all_form_fields)
                 if included_product_docs:
                     for doc in included_product_docs:
+                        decoded_doc = base64.b64decode(doc.datas)
+                        sol_id = doc_line_id_mapping[doc.id]
                         self._add_pages_to_writer(
-                            writer, base64.b64decode(doc.datas), doc_line_id_mapping[doc.id]
+                            writer, decoded_doc, all_form_fields, sol_id=sol_id
                         )
                 self._add_pages_to_writer(writer, initial_stream.getvalue())
                 if has_footer:
-                    self._add_pages_to_writer(writer, base64.b64decode(footer_record.sale_footer))
+                    decoded_footer = base64.b64decode(footer_record.sale_footer)
+                    self._add_pages_to_writer(writer, decoded_footer, all_form_fields)
 
-                form_fields = self._get_form_fields_mapping(order, doc_line_id_mapping)
-                pdf.fill_form_fields_pdf(writer, form_fields=form_fields)
+                form_fields_values_mapping = self._get_form_fields_values_mapping(
+                    order, all_form_fields, param_field_map
+                )
+                pdf.fill_form_fields_pdf(writer, form_fields=form_fields_values_mapping)
                 with io.BytesIO() as _buffer:
                     writer.write(_buffer)
                     stream = io.BytesIO(_buffer.getvalue())
                 result[order.id].update({'stream': stream})
 
         return result
 
-    def _add_pages_to_writer(self, writer, document, sol_id=None):
-        prefix = f'{sol_id}_' if sol_id else ''
+    def _add_pages_to_writer(self, writer, document, all_form_fields=None, sol_id=None):
```

Comment:
If `all_form_fields` is always given, is it necessary to specify a default value and check that it's value is not None ?

Reply:
`all_form_fields` isn't always given, not in the initial PDF, the one for the quotation itself =]